lib/src/components/NxFileUpload/types.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { InputHTMLAttributes } from 'react';
+import * as PropTypes from 'prop-types';
+
+export interface SelectedFileProps {
+  descriptionId: string;
+  file: File;
+  onDismiss: () => void;
+}
+
+// Props for NxStatefulFileUpload
+export interface StatefulProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'onChange' | 'multiple'> {
+  onChange?: ((files: FileList | null) => void) | null;
+  isRequired?: boolean | null;
+}
+
+// Props representing bits of state that need to be managed
+export interface StateProps {
+  files: FileList | null;
```

Comment:
Out of curiosity, why do we use type FileList here instead of type File, if the file upload component only accepts up to one file?

Reply:
> Out of curiosity, why do we use type FileList here instead of type File, if the file upload component only accepts up to one file?

Because `FileList` is what can be assigned to the `<input>`'s `value` property

Reply:
Ok, thanks!